testnet/stacks-node/src/tests/signer.rs

```diff
@@ -1363,13 +1239,8 @@ fn stackerdb_filter_bad_transactions() {
         .expect("Failed to write expected transactions to stackerdb");
 
     info!("------------------------- Verify Nakamoto Block Mined -------------------------");
-    let mined_block_event = signer_test.mine_nakamoto_block(timeout);
-    let hash = signer_test.wait_for_validate_ok_response(timeout);
-    let signatures = signer_test.wait_for_frost_signatures(timeout);
-    // Verify the signers accepted the proposed block and are using the previously determined dkg to sign it
-    for signature in &signatures {
-        assert!(signature.verify(&current_signers_dkg, hash.0.as_slice()));
-    }
+    let mined_block_event =
+        signer_test.mine_and_verify_confirmed_naka_block(&current_signers_dkg, timeout);
```

Comment:
A moment of truth for this PR is that once we get my downloader PR merged, we can spawn two miner runloops and verify that they can each produce blocks that the other builds atop.

Reply:
A moment of truth for this PR, or any PR that has ever consumed StackerDB events?

Reply:
Well, considering that this PR introduces a new way of sending and receiving StackerDB events, I guess both? ;)

All I was trying to say is that I wish it was possible right now to have an integration test that spanwed multiple miner runloops to verify that they can use a shared StackerDB to build blocks, but this is currently blocked for want of a downloader -- the two runloops would have no way of synchronizing their blocks in the first place.  But, I do believe it's a worthwhile integration test because it would prove that the new miner-as-signer system here works end-to-end.